components/simple-paypal.jsx

```diff
@@ -1,93 +1,30 @@
 import React from 'react';
-import { IntlMixin, FormattedMessage, FormattedNumber } from 'react-intl';
+import IntlMixin from 'react-intl';
 
-var PaypalInput = React.createClass({
-  render: function() {
-    return (
-      <div className="amount-other-wrapper">
-        <input
-          onChange={this.props.donateOnChange}
-          onKeyDown={this.props.donateKeyDown}
-          type="number" name="donation_amount_other" min={this.props.minAmount} placeholder={this.props.placeholder} className="amount-other-input medium-label-size" required value={this.props.value}/>
-      </div>
-    );
-  }
-});
-
-var PaypalButton = React.createClass({
-  mixins: [IntlMixin],
-  render: function() {
-    return (
-      <button type="submit" className="btn large-label-size" id="donate-btn">
-        { this.props.value ?
-        <FormattedMessage
-          message={this.props.message}
-          donationAmount={
-            <FormattedNumber
-              maximumFractionDigits={2}
-              value={this.props.value}
-              style="currency"
-              currency={this.props.currency || "usd"}
-            />
-          }
-          /> : this.props.defaultMessage}
-      </button>
-    );
-  }
-});
+import AmountButtons from '../components/amount-buttons.jsx';
+import Frequency from '../components/donation-frequency.jsx';
+import SubmitButton from '../components/submit-button.jsx';
+import DonateButton from '../components/donate-button.jsx';
 
 var simplePaypal = React.createClass({
-  mixins: [IntlMixin],
-  getInitialState: function() {
-    return {
-      value: ""
-    };
-  },
-  donateOnChange: function(event) {
-    this.setState({
-      value: event.target.value
-    });
-  },
-  donateKeyDown: function(event) {
-    var functionKeys = [46, 8, 9, 27, 13, 110, 190, 37, 39]; // backspace, delete, tab, escape, enter . left, and right
-    var numberKeys = [ 48, 49, 50, 51, 52, 53, 54, 55, 56, 57, 96, 97, 98, 99, 100, 101, 102, 103, 104, 105 ]; // numbers
-    var allowed = functionKeys.concat(numberKeys);
-    if (allowed.indexOf(event.keyCode) === -1) {
-      event.preventDefault();
-    }
-  },
-  componentDidMount: function() {
-    // ***********************************************
-    // Update Donate button to make it show the selected donation amount
-    // ***********************************************
-    var $theForm = $('#primary-form');
-
-    // check if a pre-selected amount is passed via the URL
-    if (window.location.hash.match(/#amount-\d+?/)) {
+  mixins: [IntlMixin, require('../mixins/form.jsx')],
+  simplePaypal(validate, props) {
```

Comment:
I kept this out of the mixin because it's only used in this file, and it was highly specific to the form data defined later in the file.